code/Controllers/CMSMain.php

```diff
@@ -1896,12 +1883,9 @@ public function performPublish($record)
      *
      * @uses SiteTree->doRevertToLive()
      *
-     * @param array $data
-     * @param Form $form
-     * @return HTTPResponse
      * @throws HTTPResponse_Exception
      */
-    public function revert($data, $form)
+    public function revert(array $data, Form $form): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
Ditto